### PR TITLE
Include info on JS transpilation with link to Babel's repl-editor

### PR DESCRIPTION
**Pull request by detroitenglish**
## Overview

This PR provides hope to developers learning of CouchDB's antique JavaScript engine, and the ancient JS dialect required therefore. 

Transpiling modern JavaScript for e.g. backwards browser compatibility is so common these days that it's arguably a fundamental skill for any front-end developer. The tooling and config that's needed, however, is NOT beginner-friendly, and so a link is provided to [Babel's in-browser real-time-transpilation text editing tool](http://babeljs.io/repl), as well as 2-step instructions on how to target the JavaScript version required by CouchDB's engine. 

In short, it informs users that modern JavaScript **can** be used, albeit with a little step in between.

## Testing recommendations

N/A

## GitHub issue number

N/A

## Related Pull Requests

N/A

## Checklist

- [ ] Documentation is written and is accurate;
- [ ] `make check` passes with no errors
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the commit hash once this PR is rebased and merged

### CODE REVIEW

**Comment by wohali** [comment]
Good idea! Most people don't need this level of JS in simple map functions, but it's a good way to work around it for people who need it.

When the Chakra code lands we won't need this advice anymore... ;)

**Comment by wohali** [comment]
Thank you for your contribution to Apache CouchDB!

**Comment by detroitenglish** [comment]
Got burned trying to use `Array.includes()` in a VDU long ago... :sweat_smile:  Really looking forward to Chakra-core landing, and thank you, Joan, for all your hard work!!
